extensions/default/src/commandsModule.ts

```diff
@@ -365,6 +374,101 @@ const commandsModule = ({
       window.setTimeout(completeLayout, 0);
     },
 
+    toggleOneUp() {
+      const viewportGridState = viewportGridService.getState();
+      const { activeViewportIndex, viewports, layout } = viewportGridState;
+      const {
+        displaySetInstanceUIDs,
+        displaySetOptions,
+        viewportOptions,
+      } = viewports[activeViewportIndex];
+
+      if (layout.numCols === 1 && layout.numRows === 1) {
+        // The viewer is in one-up. Check if there is a state to restore/toggle back to.
+        const { toggleOneUpViewportGridStore } = stateSyncService.getState();
+
+        if (toggleOneUpViewportGridStore.layout) {
+          // There is a state to toggle back to. The viewport that was
+          // originally toggled to one up was the former active viewport.
+          const viewportIndexToUpdate =
+            toggleOneUpViewportGridStore.activeViewportIndex;
+
+          // Determine which viewports need to be updated. This is particularly
+          // important for MPR.
+          const updatedViewports = displaySetInstanceUIDs
+            .map(displaySetInstanceUID =>
+              hangingProtocolService.getViewportsRequireUpdate(
+                viewportIndexToUpdate,
+                displaySetInstanceUID
+              )
+            )
+            .flat();
+
+          // This findOrCreateViewport returns either one of the updatedViewports
+          // returned from the HP service OR if there is not one from the HP service then
+          // simply returns what was in the previous state.
+          const findOrCreateViewport = (viewportIndex: number) => {
+            const viewport = updatedViewports.find(
+              viewport => viewport.viewportIndex === viewportIndex
+            );
+
+            return viewport
+              ? { viewportOptions, displaySetOptions, ...viewport }
+              : toggleOneUpViewportGridStore.viewports[viewportIndex];
+            // }
+          };
+
+          // Restore the previous layout including the active viewport.
+          viewportGridService.setLayout({
```

Comment:
please add the issue for drag and drop coronal in mpr thingy

Reply:
Certainly. I will add a separate issue once this PR is merged - before it is merged the issue does not really apply.